frontend/src/components/Gastos.js

```diff
@@ -105,6 +110,13 @@ class Gastos extends React.Component {
   gastoAdd() {
     this.listado();
   }
+
+  listadoDeGastoPorMes(event) {
+    var mes = this.state.gastos.filter(
+      item => event === moment(item.fecha).month
```

Comment:
Fijate lo que te puse en `Seleccion`: acá te está faltando pasar el parámetro para que esto ande. Otro error es que `month` es una función y vos no pusiste los paréntesis:
```suggestion
      item => event === moment(item.fecha).month()
```